fe/src/main/java/org/apache/doris/plugin/PluginMgr.java

```diff
@@ -86,25 +112,26 @@ public PluginInfo installPlugin(InstallPluginStmt stmt) throws IOException, User
 
         try {
             PluginInfo info = pluginLoader.getPluginInfo();
-
-            if (plugins[info.getTypeId()].containsKey(info.getName())) {
+            
+            if (checkDynamicPluginNameExist(info.getName())) {
```

Comment:
Using two containers does not avoid concurrency problems

Reply:
I think it works. 
I first using `checkDynamicPluginNameExist` to check if name already exist.
And after installing, I use `addDynamicPluginNameIfAbsent()` to `check-and-add` name to the `dynamicPluginNames` set. `check-and-add` is an atomic operation.

And `plugins[]` array is just for saving the plugin, I do not use it to check name existence.